### DISCUSSION TITLE
4.x 使用第三方iconfont是不是意味着必须全量导入nutui的icon?

### DISCUSSION THREAD

**Discussion by yoyo837** [jdf2e/nutui#2343] [Q&A] [answered]
因为必须这样写
```js
import { IconFont } from '@nutui/icons-vue-taro';

<IconFont font-class-name="iconfont" class-prefix="icon" name="close" />
```

**Comment by eiinu** [comment]
目前是这样的，icon 样式文件是放在 `@nutui/icons-vue-taro` 包的入口文件里的。
不过 `@nutui/nutui-taro` 里有很多组件里用到了 icon，本身就会全量导入 iconfont。

**Reply by yoyo837** [reply]
<img width="910" alt="image" src="https://github.com/jdf2e/nutui/assets/6134547/ecf10888-4fe6-4217-9698-22db32f3f4d3">

那这一步所说的按序加载能真的“按需”吗? 还是说只能在没有 `import { IconFont } ` 的前提下?

**Reply by eiinu** [reply]
这里应该是同步了 `@nutui/nutui` 的文档，使用 SVG 图标实现的按需加载。
小程序下只能使用 iconfont，应该是没有实现按需的。

**Reply by yoyo837** [reply]
不太理解“小程序下只能使用 iconfont”的意思是指什么. 小程序下不能使用 `@nutui/icons-vue-taro` 下的图标吗?

**Reply by eiinu** [reply, marked as answer]
> 不太理解“小程序下只能使用 iconfont”的意思是指什么. 小程序下不能使用 `@nutui/icons-vue-taro` 下的图标吗?

意思是文档中的使用方式是没有问题的，但是因为小程序下的图标只能采用 iconfont 方案，因此并没有实现「按需引入图标」的功能。
不管是单独引入某个 `<Add>`，还是使用 `<IconFont>`，最终结果都是全量引入字体图标。
文档已修改。

**Reply by yoyo837** [reply]
Thanks!
